### PR TITLE
Theme Detail: Add tracks event when the preview button is shown

### PR DESCRIPTION
**Pull request by rcrdortiz**
Related to:
- https://github.com/Automattic/dotcom-forge/issues/4579

## Proposed Changes

* Added an event when the `Live Preview` button is displayed.

## Testing Instructions

* Using the Calypso Live link
* Open the Theme Details page
* You should see a `calypso_live_preview_button_impression` being recorded

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by BogdanUngureanu** [comment]
The code looks fine. The unit tests are failing though.

File: client/my-sites/theme/live-preview-button/index.tsx [hunk 1]
```diff
@@ -26,8 +27,11 @@ export const LivePreviewButton: FC< Props > = ( { themeId, siteId } ) => {
 	}
 
 	return (
-		<Button onClick={ () => dispatch( livePreview( themeId, siteId, 'detail' ) ) }>
-			{ translate( 'Preview & Customize' ) }
-		</Button>
+		<>
+			<TrackComponentView eventName="calypso_live_preview_button_impression" />
```
Comment (BogdanUngureanu) [inline client/my-sites/theme/live-preview-button/index.tsx]: The corresponding click event is called `calypso_block_theme_live_preview_click`. Should we rename the event to something like `calypso_block_theme_live_preview_impression`?

**Comment by rcrdortiz** [comment]
> The code looks fine. The unit tests are failing though.

Restarting the tests did the trick.

File: client/my-sites/theme/live-preview-button/index.tsx (hunk 1 above)
Comment (rcrdortiz) [reply client/my-sites/theme/live-preview-button/index.tsx]: I kind of don't like that name but changed it anyways 🤷‍♂️

**Review by BogdanUngureanu — APPROVED** [review]
Works as expected and the event is consistent with the click one! 🐼 🚢

**Pull request merged**